### PR TITLE
added user persistence

### PR DESCRIPTION
**Pull request by danieldavidbador**

### CODE REVIEW

File: bot/database.py [hunk 1]
```diff
@@ -96,8 +116,19 @@ async def change_password(self, nickname: str, password: str, new_password: str,
...
-    async def increment_counter(self) -> int:
-        worksheet = await self._get_worksheet(1)
-        cell = await worksheet.cell(1, 2)
-        await worksheet.update_cell(1, 2, int(cell.value) + 1)
-        return int(cell.value)
\ No newline at end of file
+    async def login(self, nickname: str, password: str, userid: str) -> bool:
+        valid = await self.is_password_and_nickname_valid(nickname, password, userid)
+        if valid:
+            self.logged_in_users[userid] = nickname
```
Comment (brymer-meneses) [inline bot/database.py]: I think it would be better security-wise if we hash the userid, wdyt?

File: bot/database.py [hunk 2]
```diff
@@ -75,7 +83,7 @@ async def register_nickname(self, nickname: str, password: str, userid: str):
 
         return
     
-    async def unregister_nickname(self, nickname: str, password: str, userid: str):
+    async def delete_nickname(self, nickname: str, password: str, userid: str):
```
Comment (brymer-meneses) [inline bot/database.py]: `-> None`

File: bot/database.py [hunk 3]
```diff
@@ -49,6 +51,12 @@ async def _find_value_from_column(self, worksheet: AsyncioGspreadWorksheet, colu
                 return Cell(row_idx+2, column_number, row_value)
         return None
 
+    async def increment_counter(self) -> int:
```
Comment (brymer-meneses) [inline bot/database.py:54]: `-> None`

File: bot/database.py [hunk 4]
```diff
@@ -85,6 +93,18 @@ async def unregister_nickname(self, nickname: str, password: str, userid: str):
         await worksheet.delete_row(cell.row)
         return True
     
+    async def change_nickname(self, new_nickname: str, userid: str):
```
Comment (brymer-meneses) [inline bot/database.py]: `-> None`

**Comment by brymer-meneses** [comment]
Why did you close this? HAHAH

**Comment by danieldavidbador** [comment]
> Why did you close this? HAHAH

MAY NAMALI SA DELETE NICKNAME HAHAHAHAHHA

**Comment by danieldavidbador** [comment]
oneliner fix lang napagkaligtaan ko sorry

File: bot/database.py (hunk 1 above)
Comment (brymer-meneses) [reply bot/database.py]: Mas readable din `is_valid`

**Comment by danieldavidbador** [comment]
ohhhh automatic pala na na-aadd yung mga bagong commits?? akala ko need ko pa mag new pull request

File: bot/bot.py [hunk 5]
```diff
@@ -57,30 +72,110 @@ async def register(inter: ApplicationCommandInteraction, nickname: str, password
...
+                    description="Successfully logged out",
+                    timestamp=datetime.now(timezone))
     
-        await inter.channel.send(embed=embed)
+        await inter.edit_original_response(embed=embed)
+        return
+
+
+@bot.slash_command()
+async def user(inter):
```
Comment (brymer-meneses) [inline bot/bot.py]: Walang types?

File: bot/bot.py [hunk 6]
```diff
@@ -57,30 +72,110 @@ async def register(inter: ApplicationCommandInteraction, nickname: str, password
...
+
+    await database.delete_nickname(nickname, password, str(inter.author.id))
+    embed = Embed(title="Confession",
+                description=f"RIP 🥀 {nickname}\n\nFly high butterfly 🥹\n\n🕊️ Died peacefully in {datetime.now(timezone).strftime('%b %d, %Y')}")
+
+    await inter.channel.send(embed=embed)
+    return
+
+@user.sub_command_group()
+async def change(inter):
```
Comment (brymer-meneses) [inline bot/bot.py]: Walang types

File: bot/database.py (hunk 1 above)
Comment (danieldavidbador) [reply bot/database.py]: sige sige add ko waitteuu

File: bot/database.py (hunk 2 above)
Comment (danieldavidbador) [reply bot/database.py]: noted po

File: bot/database.py (hunk 4 above)
Comment (danieldavidbador) [reply bot/database.py]: hindi ba bool dapat?

File: bot/bot.py (hunk 5 above)
Comment (danieldavidbador) [reply bot/bot.py]: wala nga ata dapat param eh

File: bot/bot.py (hunk 6 above)
Comment (danieldavidbador) [reply bot/bot.py]: wala ng param ata dapat

Comment (danieldavidbador) [reply bot/bot.py]: https://docs.disnake.dev/en/stable/ext/commands/slash_commands.html#subcommands-and-groups

File: bot/bot.py (hunk 5 above)
Comment (danieldavidbador) [reply bot/bot.py]: https://docs.disnake.dev/en/stable/ext/commands/slash_commands.html#subcommands-and-groups

File: bot/database.py (hunk 4 above)
Comment (brymer-meneses) [reply bot/database.py]: yea my bad

File: bot/bot.py (hunk 6 above)
Comment (brymer-meneses) [reply bot/bot.py]: -> None ?

Comment (brymer-meneses) [reply bot/bot.py]: May type 'yong inter

File: bot/database.py [hunk 7]
```diff
@@ -120,16 +120,19 @@ async def change_password(self, nickname: str, password: str, new_password: str,
...
     
     async def logout(self, userid: str) -> bool:
         if userid in self.logged_in_users:
-            self.logged_in_users.pop(userid)
+            hashed_id = hashlib.sha256(bytes(userid, "utf-8"), usedforsecurity=True).hexdigest()
+            self.logged_in_users.pop(hashed_id)
             return True
         return False
     
     async def get_nickname(self, userid: str) -> Optional[str]:
```
Comment (brymer-meneses) [inline bot/database.py]: Mas descriptive pa sana 'yong `get_nickname` but I can't think of an alternative for it.

Comment (brymer-meneses) [reply bot/database.py]: `get_logged_in_nickname`?

Comment (brymer-meneses) [reply bot/database.py]: wdyt HAHAHA

File: bot/database.py (hunk 1 above)
Comment (danieldavidbador) [reply bot/database.py]: i think i've fixed all this na

File: bot/database.py (hunk 7 above)
Comment (danieldavidbador) [reply bot/database.py]: masyadong verbose?

Comment (danieldavidbador) [reply bot/database.py]: tho changed all occurences lang naman ehh HAHAHAHAHAHA

Comment (brymer-meneses) [reply bot/database.py]: `get_nickname_from_sessions`?

Comment (danieldavidbador) [reply bot/database.py]: pede naman

Comment (danieldavidbador) [reply bot/database.py]: tho session singular dapat diba?

**Comment by brymer-meneses** [comment]
LGTM!  Use `squash and merge`

**Pull request merged**